HW06.Rmd

```diff
@@ -1,92 +1,147 @@
+
+Steps: 
+
+  1. Read in the 45 processed RNA-Seq data found in "./RNA_Seq_processed"
+  2. Map gene.ids to gene.symbols (which I have for you)
+  3. For 27 genes of interest AND your favorite gene, perform a t-test to see if it is differentially expressed between the WT vs H3K27M samples
+  4. Create a graphing function and then create a boxplot that graphs expression between the two groups
+  
+  
+#### Code
+
+From the RNA-Seq files, you only need the "Name" from one file and the "TPM" column from all the files. TPM stands for "transcripts per million" and is a common unit for normalized expression data. 
+```{r load in data}
+#hint, using apply (specifically lapply) you can read in the data into a list and then bind the columns together. Or you can use a for loop too. 
+
+library(tidyverse)
+library(magrittr)
+library(dplyr)
+
+# list.files : produces a character vector of the names of files in the named directory
+H3K27M_names <- list.files("./RNA_Seq_processed/H3K27M", full.names=TRUE)
+WT_names <- list.files("./RNA_Seq_processed/WT", full.names=TRUE)
+
+#create read_in function to read in only TPM
+read_in <- function(x){
+  read_tsv(x) %>%
+  select(contains("TPM"))
+}
+
+#read in the data using read_in defined function above in the format of a data.frame
+H3K27M <- data.frame(lapply(H3K27M_names, read_in))
+WT <- data.frame(lapply(WT_names, read_in))
+
+#get name col
+name_col <- read_tsv(H3K27M_names[1]) %>%
+  select(Name) %>%
+  rename(gene_id = Name)
+
+#combine the data (list with all files)
+all_data <- c(H3K27M, WT) 
+
+#create data frame GBM.transcripts as indicated below
+GBM.transcripts <- bind_cols(name_col, all_data)
+
+
+```
+
+Now, install the packages commented below (if needed), and then use this code to map the transcript IDs to gene symbols. To use this code, you need a dataframe called GBM.transcripts that has the first column "gene_id" that contains the transcript ids (e.g. ENST00000456328.2) and the remaining columns contain the TPM data. So long as the first column contains the "gene_id" column as mentioned above, this *should* run. 
+
+```{r map gene.ids to gene.symbols using ensembldb}
+
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#BiocManager::install("EnsDb.Hsapiens.v75")
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
```

Comment:
This is a good idea, I didn't look at the table and think to remove duplicated symbols. I may try adding that into my own code!